gst/gstvalue.h

```diff
@@ -78,12 +89,13 @@ G_BEGIN_DECLS
  *
  * Can be used together with #GST_FOURCC_FORMAT to properly output a
  * #guint32 fourcc value in a printf()-style text message.
+ * For invalid fourcc substitutes non ascii chars by '?'.
  */
-#define GST_FOURCC_ARGS(fourcc) \
-        ((gchar) ((fourcc)     &0xff)), \
-        ((gchar) (((fourcc)>>8 )&0xff)), \
-        ((gchar) (((fourcc)>>16)&0xff)), \
-        ((gchar) (((fourcc)>>24)&0xff))
+#define GST_FOURCC_ARGS(fourcc) (\
+        GST_PRINTABLE_CHAR_ARGS(fourcc & 0xff), \
+        GST_PRINTABLE_CHAR_ARGS(fourcc >> 8 & 0xff), \
```

Comment:
it's a bit confusing without the branckets. I mean, what is computed first, ">>" or "&". I don't think we should expect a perfect knowledge of C standard from anybody who's reading the code. It worth it to put the brackets.

Reply:
Opinion. Operator precedence is very basic knowledge, every programmer must know it.
Extra unneeded parentheses degrades readability, not the other way around.
I don't care changing it to your preference, but I think it's better not to add extra parentheses if these are not needed.